### PR TITLE
Clarify context of SCIF app implementation

### PR DESCRIPTION
**Pull request by dtrudg**
## Description of the Pull Request (PR):

Claritfy the context of the SCIF app implementation.

Initial minimal context setting here... Having read through where/how we mention apps, and given that the SCIF implementation is and continues to be a maintained and tested part of the core code, I think we should go further toward having a separate top-level section that is more complete. We should then reference it from the usage / build / services pages.

Would love to get your thoughts @vsoch on whether this is a decent start, so we are at least guaranteed to be somewhat improved for the 3.8.0 release.

## This fixes or addresses the following GitHub issues:

- Working towards #3

### CODE REVIEW

File: definition_files.rst [hunk 1]
```diff
@@ -321,9 +321,8 @@ executed so that they are available during the build and configuration process.
 In some circumstances, it may be redundant to build different containers for
 each app with nearly equivalent dependencies. Singularity supports installing
 apps within internal modules based on the concept of `Standard Container
-Integration Format (SCI-F) <https://sci-f.github.io/>`_
-All the apps are handled by Singularity at this point. More information on
-Apps :ref:`here <apps>`.
+Integration Format (SCIF) <https://sci-f.github.io/>`_. More information on
```
Comment (vsoch) [inline definition_files.rst:324]: This is my bad - the name was ultimately the Scientific Filesystem after community discussion.

File: definition_files.rst [hunk 2]
```diff
@@ -681,16 +680,34 @@ E.g., the ``devel`` stage in the above definition cannot copy files from the
...
+SCIF is not specfic to Singularity. You can learn more about it at the
+project's site: `<https://sci-f.github.io/>`_ which includes extended
+tutorials, the specification, and other information.
+
+SCIF %app* sections
+===================
+
+SCIF apps within a Singularity container are created using ``%app*``
+sections in a definition file. These ``%app*`` sections can exist
+alongside any of the primary sections (i.e.  ``%post``,
```
Comment (vsoch) [inline definition_files.rst:705]: The one difference is post - the SCIF version would be appinstall.

Comment (vsoch) [reply definition_files.rst:705]: Same with runscript - it’s just app run. There is a complete list in the standard doc if it’s easier to direct the user there.

**Review by vsoch — COMMENTED** [review]
This is great! Thank you for the quick fixes @dtrudg.

File: definition_files.rst (hunk 2 above)
Comment (dtrudg) [reply definition_files.rst:705]: Right, those are shown in the example. The intent of this sentence...

>  "These ``%app*`` sections can exist alongside any of the primary sections (i.e.  ``%post``" i

Is that the `%app*` don't _replace_ `%post` etc. but both can be used in the same definition file. Open to rewording it.

Comment (vsoch) [reply definition_files.rst:705]: ohh gotcha, I just misread this!

File: definition_files.rst (hunk 1 above)
Comment (dtrudg) [reply definition_files.rst:324]: Gotcha - I'll check through to harmonize it. Just to confirm.... SCIF like the text on the site, or SCI-F like it is in the github URL?

Comment (vsoch) [reply definition_files.rst:324]: SCIF like the text! The one with the dash is only because the other GitHub name was taken :p

**Review by tri-adam — APPROVED** [review]
LGTM, thanks @dtrudg and @vsoch!

**Pull request merged**